### PR TITLE
string / it-academy.by / M-JC1-69-21 / Evgeni Ermakov / Done tasks 19/19 String + 3 OptionalTask

### PR DESCRIPTION
**Pull request by EvgeniErmakov**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [x] approved

### CODE REVIEW

**Review by EvgeniErmakov — COMMENTED** [review]
С выбором файлов на Pull requests еще поучусь.

File: src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+            for (String letterFromText : arrayFromText) {
+                if (letterFromText.equals(letter)) {
+                    count++;
+                }
+            }
+        }
+        return count;
+    }
+
+    public int PunctuationCounter(String text) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java:51]:
```suggestion
    public int getPunctuationCount(String text) {
```

File: src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        for (String letterFromText : arrayFromText) {
+            if (Pattern.matches("[,.?!]", letterFromText)) {
+                count++;
+            }
+        }
+
+        return count;
+    }
+
+    public boolean palindrome(String text) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java:64]:
```suggestion
    public boolean isPalindrome(String text) {
```

File: src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+        return count;
+    }
+
+    public String initialsOfTheName(String text) {
+        String[] array = text.toUpperCase().split(" ");
+
+        return array[0].substring(0, 1) + array[1].substring(0, 1);
+    }
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java:113]: Remove blank line

File: src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+        return text.replaceAll("[^0-9]", "");
+    }
+
+    public String getIndividualSymbol(String firstText, String secondText) {
+        String[] array = firstText.split("");
+        String[] array2 = secondText.split("");
+        StringBuilder result = new StringBuilder();
+
+        for (int i = 0; i < array.length; i++) {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java:125]: remove blank line

**Review by rakovets — CHANGES_REQUESTED** [review]
You need to fix the following problems

**Review by rakovets — APPROVED** [review]
Approved

**Pull request closed without merging**